src/python/pants/engine/legacy/structs.py

```diff
@@ -325,17 +337,12 @@ def from_sources_field(
 
   @staticmethod
   def _filespec_for_exclude(raw_exclude: List[str], spec_path: str) -> wrapped_globs._GlobsDict:
-    if isinstance(raw_exclude, str):
-      raise ValueError(
-        f'Excludes should be a list of strings. Got: {raw_exclude!r}'
-      )
-
     excluded_patterns: List[str] = []
     for raw_element in raw_exclude:
       exclude_filespecs = BaseGlobs.from_sources_field(raw_element, spec_path).filespecs
-      if exclude_filespecs.get('exclude', []):
+      if exclude_filespecs.get('exclude'):
```

Comment:
Why are we able to do this now?

Reply:
The `get()` now defaults to `None` instead of `[]`. This has the same end effect due to truthy values: if `exclude` is not defined, we won't enter the if block.

I made the change to have less code.

Reply:
Oh, awesome! This feels more correct anyway!